cmd/xl-storage-format-v2.go

```diff
@@ -656,82 +668,74 @@ func (z xlMetaV2) ToFileInfo(volume, path, versionID string) (fi FileInfo, err e
 		}
 	}
 
-	var latestModTime time.Time
-	var latestIndex int
-	for i, version := range z.Versions {
+	for _, version := range z.Versions {
 		if !version.Valid() {
 			logger.LogIf(GlobalContext, fmt.Errorf("invalid version detected %#v", version))
-			return FileInfo{}, errFileNotFound
-		}
-		var modTime time.Time
-		switch version.Type {
-		case ObjectType:
-			modTime = time.Unix(0, version.ObjectV2.ModTime)
-		case DeleteType:
-			modTime = time.Unix(0, version.DeleteMarker.ModTime)
-		case LegacyType:
-			modTime = version.ObjectV1.Stat.ModTime
-		}
-		if modTime.After(latestModTime) {
-			latestModTime = modTime
-			latestIndex = i
+			if versionID == "" {
+				return FileInfo{}, errFileNotFound
+			}
+			return FileInfo{}, errFileVersionNotFound
+
 		}
 	}
 
+	sort.Sort(xlMetaV2VersionsSorter(z.Versions))
```

Comment:
You modify `z.Versions` on a value receiver. The caller wouldn't expect that.

Reply:
> You modify `z.Versions` on a value receiver. The caller wouldn't expect that.

Created a new slice and copied z.Versions there